vignettes/printslides.Rmd

```diff
@@ -0,0 +1,232 @@
+---
+title: "Print complex `{xaringan}` slides"
+author: "Bernhard Bieri"
+date: "`r Sys.Date()`"
+
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Print complex `{xaringan}` slides}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r global.options, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.align = 'center',
+  fig.width=6,
+  fig.height=4
+)
+```
+
+Printing slides created with the `{iheiddown}` presentation template works
+fine in 90% of the time with an internet browser's print function. However,
+for more complex layouts, such as revealing slides, one may want greater
+control over the printing output. This article, and the function below,
+aim at solving this issue.
+
+The function itself is borrowed from Garrick Aden Buie's excellent
+[blog](https://www.garrickadenbuie.com/blog/print-xaringan-chromote/) and
+leverages `{chromote}`, a headless version of the Chrome web browser to flick
+through every slide automatically and print them out one by one.
+
+Note that this function is not included into `{iheiddown}` package as it relies
+on the following packages (some of which are not on CRAN yet): `{chromote}`,
+`{progress}`, `{jsonlite}`, `{pdftools}`, and `{digest}`.
+
+## The workflow:
+
+1. Open RStudio and the HTML version of the xaringan presentation
+
+2. Copy paste the function below into your session and add it to your global
+    environment
+
+3. Run `xaringan_to_pdf()` with the path to your original HTML presentation
+    and specify an `output_file` path where you want the PDF version of your
+    presentation to be printed.
+
+4. Wait for the function to flick through all the slides and check out the
+   result!
+
+
+```{r PrintFunction, eval=FALSE}
+#' Print xaringan slides to PDF
+#'
+#' Prints xaringan slides to a PDF file, even complicated slides
+#' with panelsets or other html widgets or advanced features.
+#' Requires a local installation of Chrome as well as the following packages:
+#' `{chromote}`, `{progress}`, `{jsonlite}`, `{pdftools}`, and `{digest}`.
+#'
+#' @param input Path to Rmd or html file of xaringan slides.
+#' @param output_file The name of the output file. If using NULL then
+#'   the output filename will be based on filename for the input file.
+#'   If a filename is provided, a path to the output file can also be provided.
+#' @param delay Seconds of delay between advancing to and printing
+#'   a new slide.
+#' @param include_partial_slides Should partial (continuation) slides be
+#'   included in the output? If `FALSE`, the default, only the complete slide
+#'   is included in the PDF.
+#' @source <https://www.garrickadenbuie.com/blog/print-xaringan-chromote/>
+
+xaringan_to_pdf <- function(
```

Comment:
This is a large, complex function to have in a vignette. Perhaps just incorporate it into the package proper?

Reply:
Agreed, the function comes from Garrick Aden‑Buie's blog and uses the `{chromote}` which as of now is not available on CRAN. While I thought of integrating it into our package as well, this would lead CRAN to refuse our package.  Garrick hasn't integrated it in any package for that exact reason too so I thought that this was a good compromise between the need of having such a printing function readily available and having our package on CRAN.

Reply:
Ok, do we need to have the whole function here then? Can't we just suggest/recommend users to download and use the function from Garrick's package?

Reply:
We can, that's basically what we had before. I just thought it might be too buried being at the end of the presentation vignette so I highlighted it in its own vignette. However, I do agree with you that it looks cleaner without having an extra vignette for this only.

I reverted the changes and it is now linked under the "printing your slides" section. I also made a note to myself to include this in the upcoming troubleshooting vignette so that it won't get overlooked by users.